### PR TITLE
Add arm64 support on MacOS

### PR DESCRIPTION
**Pull request by q7zh**
- Add arm64 support on macOS.
- Updated to use macOS 10.12 as the minimum OS

### CODE REVIEW

File: RadeonRays/CMakeLists.txt [hunk 1]
```diff
@@ -232,10 +232,10 @@ if (UNIX AND NOT APPLE)
         string(REGEX MATCH "[0-9]*\.[0-9]*$" RR_API_VERSION ${RR_API_VERSION})
         
         set_target_properties(RadeonRays PROPERTIES SOVERSION ${RR_API_VERSION})
-elseif (APPLE AND CMAKE_OSX_ARCHITECTURES=x86_64)
-    target_compile_options(RadeonRays PUBLIC -stdlib=libc++ -msse4.1 -mmacosx-version-min=10.12)
 elseif (APPLE AND CMAKE_OSX_ARCHITECTURES=arm64)
     target_compile_options(RadeonRays PUBLIC -stdlib=libc++ -mmacosx-version-min=10.12)
```
Comment (KEngelstoft) [inline RadeonRays/CMakeLists.txt]: Apple Silicon is only supported on Big Sur and later (10.16 or 11.0), -mmacosx-version-min=10.12 doesn't make sense.

File: RadeonRays/src/accelerator/sse2neon.h [hunk 2]
```diff
@@ -0,0 +1,6411 @@
+#ifndef SSE2NEON_H
+#define SSE2NEON_H
+
+// This header file provides a simple API translation layer
+// between SSE intrinsics to their corresponding Arm/Aarch64 NEON versions
+//
+// This header file does not yet translate all of the SSE intrinsics.
+//
+// Contributors to this work are:
```
Comment (KEngelstoft) [inline RadeonRays/src/accelerator/sse2neon.h]: Where did this file come from? A link to a repo and its revision would be nice.

File: RadeonRays/src/accelerator/sse2neon.h [hunk 3]
```diff
@@ -0,0 +1,6411 @@
+#ifndef SSE2NEON_H
+#define SSE2NEON_H
+
+// This header file provides a simple API translation layer
+// between SSE intrinsics to their corresponding Arm/Aarch64 NEON versions
```
Comment (KEngelstoft) [inline RadeonRays/src/accelerator/sse2neon.h]: I think this file should live in 3rdparty folder, e.g. 3rdparty/sse2neon

**Review by KEngelstoft — CHANGES_REQUESTED** [review]
Please fix the min macOS version, add more info regarding sse2neon and move the file.

File: RadeonRays/CMakeLists.txt (hunk 1 above)
Comment (q7zh) [reply RadeonRays/CMakeLists.txt]: Fix the min macOS version to 11.0 for the arm64 build. For the x86_64 build, 10.12 is what Apple suggests to be the min OS version now.

File: RadeonRays/src/accelerator/sse2neon.h (hunk 2 above)
Comment (q7zh) [reply RadeonRays/src/accelerator/sse2neon.h]: Added the repo link and revision in the comment and moved the file to 3rdparty folder.

**Comment by q7zh** [comment]
Fixed the issues mentioned above. Please hold this PR until we have a legal sign off for sse2neon.

**Review by KEngelstoft — APPROVED** [review]

**Comment by KEngelstoft** [comment]
Are the -mmacosx-version-min= directives redundant now?

**Comment by q7zh** [comment]
Right, it seems it is not taking any effect. I will remove it.

**Pull request merged**